bal_tools/subgraph.py

```diff
@@ -62,50 +62,56 @@ def get_subgraph_url(self, subgraph="core") -> str:
         elif subgraph == "gauges":
-            magic_word = "gaugesSubgraph:"
+            magic_word = "gauge:"
         elif subgraph == "blocks":
-            magic_word = "blockNumberSubgraph:"
+            magic_word = "blocks:"
         elif subgraph == "aura":
             return AURA_SUBGRAPHS_BY_CHAIN.get(self.chain, None)
 
-        # get subgraph url from sdk config
-        sdk_file = f"https://raw.githubusercontent.com/balancer/balancer-sdk/develop/balancer-js/src/lib/constants/config.ts"
+        # get subgraph url from frontend config
+        chain_url_slug = 'gnosis-chain' if self.chain == 'gnosis' else self.chain
+        config_file = f"https://raw.githubusercontent.com/balancer/frontend-v2/develop/src/lib/config/{chain_url_slug}/index.ts"
         found_magic_word = False
-        urls_reached = False
-        with urlopen(sdk_file) as f:
+        with urlopen(config_file) as f:
             for line in f:
-                if "[Network." in str(line):
-                    chain_detected = str(line).split("[Network.")[1].split("]")[0].lower()
-                    if chain_detected == self.chain:
-                        for line in f:
-                            if "urls: {" in str(line) or urls_reached:
-                                urls_reached = True
-                                if "}," in str(line):
-                                    return None
-                                if found_magic_word:
-                                    url = (
-                                        line.decode("utf-8")
-                                        .strip()
-                                        .split(",")[0]
-                                        .strip(" ,'")
-                                    )
-                                    url = re.sub(
-                                        r"(\s|\u180B|\u200B|\u200C|\u200D|\u2060|\uFEFF)+",
-                                        "",
-                                        url,
-                                    )
-                                    if urlparse(url).scheme in [
-                                        "http",
-                                        "https",
-                                    ]:
-                                        return url
-                                    return None
-                                if magic_word in str(line):
-                                    # url is on next line, return it on the next iteration
-                                    found_magic_word = True
-            return None
+                if found_magic_word or magic_word + " `" in str(line):
+                    # url is on this line
+                    r = re.search('`(.*)`', line.decode("utf-8"))
+                    try:
+                        url = r.group(1)
+                        if urlparse(url).scheme in ["http", "https"]:
+                            return url.replace('${keys.graph}', os.getenv("GRAPH_API_KEY"))
```

Comment:
this will raise a type error if there is no graph key. should this be handled gracefully so it can fallback to legacy subgraph or is this intended?

Reply:
56aff64